src/jobservice/job/workerpool.go

```diff
@@ -15,29 +15,35 @@
 package job
 
 import (
-	"github.com/vmware/harbor/src/common/dao"
+	"fmt"
+
 	"github.com/vmware/harbor/src/common/models"
 	"github.com/vmware/harbor/src/common/utils/log"
 	"github.com/vmware/harbor/src/jobservice/config"
 )
 
+// workerPool is a set of workers each worker is associate to a statemachine for handling jobs.
+// it consists of a channel for free workers and a list to all workers
 type workerPool struct {
+	poolType   Type
 	workerChan chan *Worker
 	workerList []*Worker
 }
 
-// WorkerPool is a set of workers each worker is associate to a statemachine for handling jobs.
-// it consists of a channel for free workers and a list to all workers
-var WorkerPool *workerPool
+// WorkerPools is a map contains workerpools for different types of jobs.
+var WorkerPools map[Type]*workerPool
```

Comment:
Why we need two kinds of worker pool? Should not the workers be same as they all handle jobs? The handlers of jobs can be different.